src/uct/sm/cma/cma_pd.c

```diff
@@ -0,0 +1,59 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+* Copyright (c) UT-Battelle, LLC. 2014-2015. ALL RIGHTS RESERVED.
+*
+* $COPYRIGHT$
+* $HEADER$
+*/
+
+#include "cma_pd.h"
+
+ucs_status_t uct_cma_pd_query(uct_pd_h pd, uct_pd_attr_t *pd_attr)
+{
+    pd_attr->rkey_packed_size  = sizeof(uct_cma_packed_rkey_t);
+    pd_attr->cap.flags         = UCT_PD_FLAG_REG;
+    pd_attr->cap.max_alloc     = 0;
+    pd_attr->cap.max_reg       = ULONG_MAX;
+
+    memset(&pd_attr->local_cpus, 0xff, sizeof(pd_attr->local_cpus));
+    return UCS_OK;
+}
+
+ucs_status_t uct_cma_mkey_pack(uct_pd_h pd, uct_mem_h memh, void *rkey_buffer)
+{
+    uct_cma_packed_rkey_t *rkey = rkey_buffer;
+
+    rkey->cma_id      = getpid();
```

Comment:
@shamisp as we discussed i think pid() should be part of iface address and not rkey